### ISSUE TITLE
Patch file for RN 0.65.+

### ISSUE THREAD

**Issue by kanzitelli** [Kureev/react-native-blur#444]
Since React Native is now officially 0.65.+ version, `jcenter()` was removed from the Android part.
This library relies on [Dimezis/BlurView](https://github.com/Dimezis/BlurView) for Android and the library uses https://jitpack.io/ as stated [here](https://github.com/Dimezis/BlurView#gradle).

So while updating RN for [rnn-starter](https://github.com/kanzitelli/rnn-starter), I have come up with the patch file which you can use with [patch-package](https://github.com/ds300/patch-package).

`@react-native-community+blur+3.6.0.patch`
```
diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
index 8177235..5d310b9 100644
--- a/node_modules/@react-native-community/blur/android/build.gradle
+++ b/node_modules/@react-native-community/blur/android/build.gradle
@@ -9,7 +9,6 @@ buildscript {
     if (project == rootProject) {
         repositories {
             google()
-            jcenter()
         }
 
         dependencies {
@@ -37,11 +36,12 @@ android {
 
 repositories {
     google()
-    jcenter()
+
+    maven { url 'https://jitpack.io' }
 }
 
 dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
-    implementation 'com.eightbitlab:blurview:1.6.3'
+    implementation 'com.github.Dimezis:BlurView:version-1.6.6'
 }
```

Link to [Gist](https://gist.github.com/kanzitelli/e79bab6dbd6b5117c879eeab31725bf0).
And here are [available versions](https://jitpack.io/#Dimezis/BlurView) of BlurView on jitpack.

**Comment by LadislavBohm** [comment]
Should we consider contacting owner of the repo to add more contributors? This package seems to used a lot with almost 2 years of inactivity.

**Comment by SGarcia710** [comment]
> Since React Native is now officially 0.65.+ version, `jcenter()` was removed from the Android part.
> This library relies on [Dimezis/BlurView](https://github.com/Dimezis/BlurView) for Android and the library uses https://jitpack.io/ as stated [here](https://github.com/Dimezis/BlurView#gradle).
> 
> So while updating RN for [rnn-starter](https://github.com/kanzitelli/rnn-starter), I have come up with the patch file which you can use with [patch-package](https://github.com/ds300/patch-package).
> 
> `@react-native-community+blur+3.6.0.patch`
> 
> ```
> diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> index 8177235..5d310b9 100644
> --- a/node_modules/@react-native-community/blur/android/build.gradle
> +++ b/node_modules/@react-native-community/blur/android/build.gradle
> @@ -9,7 +9,6 @@ buildscript {
>      if (project == rootProject) {
>          repositories {
>              google()
> -            jcenter()
>          }
>  
>          dependencies {
> @@ -37,11 +36,12 @@ android {
>  
>  repositories {
>      google()
> -    jcenter()
> +
> +    maven { url 'https://jitpack.io' }
>  }
>  
>  dependencies {
>      //noinspection GradleDynamicVersion
>      implementation 'com.facebook.react:react-native:+'
> -    implementation 'com.eightbitlab:blurview:1.6.3'
> +    implementation 'com.github.Dimezis:BlurView:version-1.6.6'
>  }
> ```
> 
> Link to [Gist](https://gist.github.com/kanzitelli/e79bab6dbd6b5117c879eeab31725bf0).
> And here are [available versions](https://jitpack.io/#Dimezis/BlurView) of BlurView on jitpack.

How can I execute this?

**Comment by chungweileong94** [comment]
> > Since React Native is now officially 0.65.+ version, `jcenter()` was removed from the Android part.
> > This library relies on [Dimezis/BlurView](https://github.com/Dimezis/BlurView) for Android and the library uses https://jitpack.io/ as stated [here](https://github.com/Dimezis/BlurView#gradle).
> > So while updating RN for [rnn-starter](https://github.com/kanzitelli/rnn-starter), I have come up with the patch file which you can use with [patch-package](https://github.com/ds300/patch-package).
> > `@react-native-community+blur+3.6.0.patch`
> > ```
> > diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> > index 8177235..5d310b9 100644
> > --- a/node_modules/@react-native-community/blur/android/build.gradle
> > +++ b/node_modules/@react-native-community/blur/android/build.gradle
> > @@ -9,7 +9,6 @@ buildscript {
> >      if (project == rootProject) {
> >          repositories {
> >              google()
> > -            jcenter()
> >          }
> >  
> >          dependencies {
> > @@ -37,11 +36,12 @@ android {
> >  
> >  repositories {
> >      google()
> > -    jcenter()
> > +
> > +    maven { url 'https://jitpack.io' }
> >  }
> >  
> >  dependencies {
> >      //noinspection GradleDynamicVersion
> >      implementation 'com.facebook.react:react-native:+'
> > -    implementation 'com.eightbitlab:blurview:1.6.3'
> > +    implementation 'com.github.Dimezis:BlurView:version-1.6.6'
> >  }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Link to [Gist](https://gist.github.com/kanzitelli/e79bab6dbd6b5117c879eeab31725bf0).
> > And here are [available versions](https://jitpack.io/#Dimezis/BlurView) of BlurView on jitpack.
> 
> How can I execute this?

You need to use this https://www.npmjs.com/package/patch-package

**Comment by gcrozariol** [comment]
> > > Since React Native is now officially 0.65.+ version, `jcenter()` was removed from the Android part.
> > > This library relies on [Dimezis/BlurView](https://github.com/Dimezis/BlurView) for Android and the library uses https://jitpack.io/ as stated [here](https://github.com/Dimezis/BlurView#gradle).
> > > So while updating RN for [rnn-starter](https://github.com/kanzitelli/rnn-starter), I have come up with the patch file which you can use with [patch-package](https://github.com/ds300/patch-package).
> > > `@react-native-community+blur+3.6.0.patch`
> > > ```
> > > diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> > > index 8177235..5d310b9 100644
> > > --- a/node_modules/@react-native-community/blur/android/build.gradle
> > > +++ b/node_modules/@react-native-community/blur/android/build.gradle
> > > @@ -9,7 +9,6 @@ buildscript {
> > >      if (project == rootProject) {
> > >          repositories {
> > >              google()
> > > -            jcenter()
> > >          }
> > >  
> > >          dependencies {
> > > @@ -37,11 +36,12 @@ android {
> > >  
> > >  repositories {
> > >      google()
> > > -    jcenter()
> > > +
> > > +    maven { url 'https://jitpack.io' }
> > >  }
> > >  
> > >  dependencies {
> > >      //noinspection GradleDynamicVersion
> > >      implementation 'com.facebook.react:react-native:+'
> > > -    implementation 'com.eightbitlab:blurview:1.6.3'
> > > +    implementation 'com.github.Dimezis:BlurView:version-1.6.6'
> > >  }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Link to [Gist](https://gist.github.com/kanzitelli/e79bab6dbd6b5117c879eeab31725bf0).
> > > And here are [available versions](https://jitpack.io/#Dimezis/BlurView) of BlurView on jitpack.
> > 
> > 
> > How can I execute this?
> 
> You need to use this https://www.npmjs.com/package/patch-package

So, I installed the patch-package, but I'm not sure how to run this command. 

I tried nom patch-package @react-native-community+blur+3.6.0.patch, but I guess that's not correct. 

Could you please explain how to do this patching? 

Thanks!

**Comment by LadislavBohm** [comment]
Don't be lazy and read docs for patch-package.

**Comment by kanzitelli** [comment]
@gcrozariol you can check out the [rnn-starter](https://github.com/kanzitelli/rnn-starter) that i maintain and use this library with patch-package

**Comment by gcrozariol** [comment]
> Don't be lazy and read docs for patch-package.

Well... I was reading ir, I just didn't get it. Have never heard about it.

**Comment by gcrozariol** [comment]
> @gcrozariol you can check out the [rnn-starter](https://github.com/kanzitelli/rnn-starter) that i maintain and use this library with patch-package

Oh! Now I see what you did there. Thanks, dude!

**Comment by kanzitelli** [comment]
So, for those who struggle with running this patch, here is a small tutorial:

1. Create `patches` folder in a project's root.
2. Create `@react-native-community+blur+3.6.0.patch` file there and fill it in as shown above.
3. Run `yarn add -D patch-package postinstall-postinstall`.
4. Add `"postinstall": "patch-package"` to scripts section in `package.json`.
5. Now after running `yarn add/remove/etc.`, the patch file will be applied to the corresponding library.

For more information and example, check out [rnn-starter](https://github.com/kanzitelli/rnn-starter).

**Comment by nriccar** [comment]
thanks!!

**Comment by hassanQureshi22** [comment]
This isn't working I am getting this error

![image](https://user-images.githubusercontent.com/24655331/135826212-77851fa6-4903-4143-8d53-e120356cc3fc.png)

**Comment by chungweileong94** [comment]
> This isn't working I am getting this error
> 
> ![image](https://user-images.githubusercontent.com/24655331/135826212-77851fa6-4903-4143-8d53-e120356cc3fc.png)

I'm not 100% sure, but I don't think this is something related to this issue.
Technically, you can still use `jcenter()` without the `patch-package`.

**Comment by hassanQureshi22** [comment]
I am getting the same error when adding jcenter() in build.gradle

**Comment by hassanQureshi22** [comment]
Well I have fix this issue by adding this in app/build.gradle

![image](https://user-images.githubusercontent.com/24655331/135829735-7abc6351-5901-44cf-a546-5172031bcaaf.png)

**Comment by hassanQureshi22** [comment]
> This isn't working I am getting this error
> 
> ![image](https://user-images.githubusercontent.com/24655331/135826212-77851fa6-4903-4143-8d53-e120356cc3fc.png)

Well I have fix this issue by adding this in app/build.gradle

Well I have fix this issue by adding this in app/build.gradle

**Comment by chrisliiingo** [comment]
> Don't be lazy and read docs for patch-package.

@LadislavBohm Please don't talk to people like that. 

@Kureev can you delete that comment please.

**Comment by fiznool** [comment]
If you are only using this package on iOS, you can prevent it being autolinked on Android by adding the following to the file `react-native.config.js` inside the root of your project:

``` js
module.exports = {
  dependencies: {
    '@react-native-community/blur': {
      platforms: {
        android: null,
      },
    },
  },
};
```

If you do this, ensure that the import of this package happens in an iOS platform-specific file [as explained in the React Native docs](https://reactnative.dev/docs/platform-specific-code#platform-specific-extensions).

**Comment by Titozzz** [comment]
fixed in 4.0.0
